Scripts/RemediationScripts/Remediate-SetDBForMySQLFlexibleServerMinReqTLSVersion.ps1

```diff
@@ -334,9 +334,9 @@ function Set-SecureTLSVersionForDBForMySQLFlexibleServer {
                     $DBForMySQLFSResource = Get-AzMySqlFlexibleServer -ResourceGroupName $_.ResourceGroupName -Name $_.ResourceName -ErrorAction SilentlyContinue
             
                     $parameterValue = (Get-AzMySqlFlexibleServerConfiguration -Name $([Constants]::ParameterName) -ResourceGroupName $_.ResourceGroupName -ServerName $_.ResourceName -SubscriptionId $SubscriptionId).Value
-                    $DBForMySQLFlexibleServerDetails += $_  | Select-Object @{N = 'ResourceId'; E = { $_.Id } },
-                    @{N = 'ResourceGroupName'; E = { $_.Id.Split("/")[4] } },
-                    @{N = 'ResourceName'; E = { $_.Name } }, 
+                    $DBForMySQLFlexibleServerDetails += $_  | Select-Object @{N = 'ResourceId'; E = { $_.ResourceId } },
+                    @{N = 'ResourceGroupName'; E = { $_.ResourceGroupName } },
```

Comment:
Same changes should be done to the auto remediation mode code flow to keep things uniform.

Reply:
I have done the same change as we are doing with auto remediation and from the subscription as well. Using ID and splitting it to get the RG name now in both the condition.